# Morinkit: Python functions for cancer genomics

[![PyPI](https://img.shields.io/pypi/v/morinkit.svg)](https://pypi.python.org/pypi/morinkit)
[![Build Status](https://img.shields.io/travis/morinlab/morinkit/master.svg)](https://travis-ci.org/morinlab/morinkit)
[![Coverage Status](https://img.shields.io/codecov/c/github/morinlab/morinkit/master.svg)](https://codecov.io/gh/morinlab/morinkit/branch/master)

Morinkit is a Python package that bundles several functions meant for cancer genomics. Examples include:

* **ensemble_vcf:** Perform ensemble SNV calling by integrating VCF files generated by different methods. 
* **augment_maf (coming soon):** Annotate a mutation annotation file with the allele counts from various BAM files.

## Installation

You need Python 3.3 or later to install morinkit. 

```bash
pip install numpy Cython  # Or `conda install numpy Cython`
pip install morinkit
```

## Dependencies

* Python 3.3 or later
* cyvcf2
* pysam

## Functions

**ensemble_vcf:** One of the ICGC-TCGA DREAM challenges showed that ensemble SNV calling led to the highest accuracy (doi:10.1038/nmeth.3407). Ensemble_vcf is primarily meant to integrate the VCF files from different methods and only output variants supported by some number of methods (or fraction of methods).
